docs/user-documentation/file_viewers.md

```diff
@@ -0,0 +1,84 @@
+# File Viewers
+
+_last updated on 03-12-2020_
+
+
+## What are viewers?
+
+[Viewers](../glossary#viewer) allow site builders to display files in interactive javascript-based widgets, that provide functionality like zooming in/out, turning pages, playing/pausing, viewing in full screen, etc. 
+
+In Drupal, a common way to implement a viewer is through a [module](./glossary/#module) that provides the javascript library, and a field formatter that uses that library. Usually the field formatter can be used on File fields.  
+
+Viewers that are known to work with Islandora 8 include:
+
+* [OpenSeadragon](https://openseadragon.github.io/), via the Drupal module [OpenSeadragon](https://github.com/Islandora/openseadragon) (maintained by the Islandora Foundation)
+* [pdf.js](https://github.com/mozilla/pdf.js), via the Drupal contrib module [PDF](https://www.drupal.org/project/pdf)
+
+Both are included with Islandora Defaults.
+
+## How are viewers configured?
+
+In the generic Islandora 8 content modelling scenario, we often want the viewer to appear on a _"resource node"_ when the file itself is in a field on a Media that references that node. These layers of indirection make displaying viewers on nodes rather complex.
+
+## Example - Open Seadragon (in Islandora Defaults)
+
+Here is how viewers are implemented in Islandora Defaults. This example uses OpenSeadragon, but pdf.js is configured with the same method. This method was chosen because it uses pure configuration and no extra code. It relies on users tagging nodes with appropriate "display hint" terms to trigger Contexts to switch the node's view mode to one that includes an EVA field, which renders the appropriate Media's file in the viewer.
+
+### Expected behaviour for this scenario:
+
+Create a Repository Item, and tag it with "Open Seadragon" (under Display Hints). Add a Media of type "Image" to that node, and tag it with "Original File" (under Media Use). Upload an image file that can be viewed in OpenSeadragon, for example, a jpeg-2000 file, and save (publish) the Media. Go to the Repository Item's page, and you should see your file displayed in the viewer.
+
+
+### Components of this scenario: 
+
+1. A viewer (e.g. the module OpenSeadragon)
+1. A media display mode (e.g. the media display mode "Open Seadragon")
+1. A media type (see Image, or also File.)
+1. A file field on that media type that can use that viewer as a field formatter (e.g. field_media_image)
```

Comment:
Might be more accurately written as "that is supported by the viewer's field formatter".

Reply:
I'm legitimately curious, what is the difference here? 

Does the _actual file_ come into play in terms of being supported by the field formatter? I see that that is relevant to this working - e.g. I"m pretty sure the File media type accepts, in field_media_file, file types that don't work with OpenSeadragon and/or Cantaloupe.

In this case I'm talking about the way that OpenSeadragon only works on certain field types, e.g. you can't use it on a "Long text with summary".

Reply:
The module that provides the viewer includes a FieldFormatter plugin and that plugin's definition includes what field types it supports. E.g. [OpenSeadragon supports both image and file fields](https://github.com/Islandora/openseadragon/blob/8.x-1.x/src/Plugin/Field/FieldFormatter/OpenSeadragonImageFormatter.php#L18-L20) whereas [PDF.js only supports file fields](https://git.drupalcode.org/project/pdf/-/blob/8.x-1.0/src/Plugin/Field/FieldFormatter/PdfDefault.php#L15). 

So, what matters, first of all, is the type of file field (file, image, etc.) which determines which formatters are available and, secondly, the actual file itself. (E.g. you could store a Jpeg in a file field, but the PDF.js viewer will probably throw an error.) Does that help?

Reply:
thanks!